airflow/configuration.py

```diff
@@ -469,11 +469,22 @@ class AirflowConfigParser(ConfigParser):
     }
 
     def __init__(self, *args, **kwargs):
+        default_config = kwargs.pop('default_config', None)
         ConfigParser.__init__(self, *args, **kwargs)
-        self.read_string(parameterized_config(DEFAULT_CONFIG))
+        self.defaults = ConfigParser(*args, **kwargs)
+
+        AirflowConfigParser._read_string(
+            self.defaults, parameterized_config(DEFAULT_CONFIG))
+        if default_config:
+            AirflowConfigParser._read_string(
+                self.defaults, parameterized_config(default_config))
         self.is_validated = False
 
     def read_string(self, string, source='<string>'):
+        AirflowConfigParser._read_string(self, string, source)
+
+    @staticmethod
```

Comment:
I got a `TypeError` when I use `AirflowConfigParser.read_string(self.default, default_config)`. The function required `AirflowConfigParser` instance, but `default_config` object is `ConfigParser` instance.
To avoid this, I make it static.

Is there a good pattern to avoid this?